services/todo-legacy/repositories/todo.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AnyObject,
+  DataSourceConstructor,
+  DefaultCrudRepository,
+  Filter,
+  Model,
+  ModelBaseConstructor,
+  ObjectType,
+  Repository,
+  Where,
+  juggler
+} from '@loopback/repository';
+import { inject, Application } from '@loopback/core';
+
+export class TodoRepository {
+  // FIXME(kev): The juggler.ModelBase type doesn't carry any definitions
+  // for the CRUD functions that typically get mixed in when createModel
+  // makes a definition, so we have to use type "any" to avoid complaint!
+  model: juggler.ModelBase | any;
+  datasource: juggler.DataSource;
+  constructor(
+    @inject('datasources.config') config: Object,
+    @inject('models.todo') modelDefinition: any
+  ) {
+    this.datasource = new DataSourceConstructor(config);
+    this.model = this.datasource.createModel(
+      modelDefinition.name,
+      modelDefinition.properties
+    );
+  }
+
+  // API definitions
+  async find(id?: number, filter?: Filter) {
+    return id ? await this.model.findById(id) : await this.model.find(filter);
+  }
+
+  async create(model: any): Promise<any> {
+    return await this.model.create(model);
+  }
+
+  async update(id: number, model: any): Promise<any> {
+    if (!id) return Promise.reject(new Error('You must provide an id!'));
+    model.id = id;
+    var result = await this.model.findById(id);
+    return await result.updateAttributes(model);
```

Comment:
FWIW, this has a race condition. It's the same problem we have in LoopBack 3.x, but it's more visible here. Perhaps add a comment to point this out?